rpc_util.go

```diff
@@ -166,6 +167,15 @@ type afterCall func(c *callInfo)
 func (o afterCall) before(c *callInfo) error { return nil }
 func (o afterCall) after(c *callInfo)        { o(c) }
 
+// CallMetadata returns a CallOption that will attach the given metadata
+// to the call headers.
+func CallMetadata(k, v string) CallOption {
```

Comment:
The CallOption returned by this function should be magical and allow introspection, so interceptors can remove unwanted metadata if necessary.  It should be able to return `k` and `v` (or `kv []string`).

Reply:
What's the method by which interceptors do this at the moment? Is there an interface to conform to, or is it as simple as defining a new type of CallOption that has some `GetMetadata` method receiver on it? (Please feel free to guide me towards a doc if I'm missing it! Poked around a bit but couldn't see anything)

Reply:
Actually, let's leave this functionality out for this PR.  We are discussing options for making this work for all our `CallOption`s, not just this new one.  We have some ideas floating around, but haven't decided on anything yet...

Reply:
👍  sgtm